mongo/gridfs/gridfs_test.go

```diff
@@ -35,8 +35,16 @@ func TestGridFS(t *testing.T) {
 			}
 		},
 	}
-	clientOpts := options.Client().ApplyURI(cs.Original).SetReadPreference(readpref.Primary()).
-		SetWriteConcern(writeconcern.New(writeconcern.WMajority())).SetPoolMonitor(poolMonitor)
+	clientOpts := options.Client().
+		ApplyURI(cs.Original).
+		SetReadPreference(readpref.Primary()).
+		SetWriteConcern(writeconcern.New(writeconcern.WMajority())).
+		SetPoolMonitor(poolMonitor).
+		// Connect to a single host. For sharded clusters, this will pin to a single mongos, which avoids
+		// non-deterministic versioning errors in the server. This has no effect for replica sets because the driver
+		// will discover the other hosts during SDAM checks.
```

Comment:
Right, and `directConnection=false` is not necessary because a single host in Go still initializes the topology type to `Unknown`, right? (libmongoc initializes to `Single`).

Reply:
Yes, the default is to discover the whole replica set even if only one member is provided.